Makefile

```diff
@@ -280,55 +286,58 @@ stop-etcd:
 run-kubernetes-master: stop-kubernetes-master
 	# Run a Kubernetes apiserver using Docker.
 	docker run \
-		--net=host --name st-apiserver \
+		--net=host --name st-apiserver-${KUBE_MASTER_PORT} \
 		--detach \
 		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kube-apiserver \
 			--bind-address=0.0.0.0 \
+			--secure-port=1${KUBE_MASTER_PORT} \
 			--insecure-bind-address=0.0.0.0 \
+			--port=${KUBE_MASTER_PORT} \
 	        	--etcd-servers=http://127.0.0.1:2379 \
 			--admission-control=NamespaceLifecycle,LimitRanger,DefaultStorageClass,ResourceQuota \
 			--service-cluster-ip-range=10.101.0.0/16 \
 			--v=10 \
 			--logtostderr=true
 
 	# Wait until the apiserver is accepting requests.
-	while ! docker exec st-apiserver kubectl get nodes; do echo "Waiting for apiserver to come up..."; sleep 2; done
+	while ! docker exec st-apiserver-${KUBE_MASTER_PORT} kubectl get nodes; do echo "Waiting for apiserver to come up..."; sleep 2; done
 
 	# And run the controller manager.
 	docker run \
-		--net=host --name st-controller-manager \
+		--net=host --name st-controller-manager-${KUBE_MASTER_PORT} \
 		--detach \
 		gcr.io/google_containers/hyperkube-amd64:${K8S_VERSION} kube-controller-manager \
-                        --master=127.0.0.1:8080 \
+                        --master=127.0.0.1:${KUBE_MASTER_PORT} \
                         --min-resync-period=3m \
                         --allocate-node-cidrs=true \
                         --cluster-cidr=10.10.0.0/16 \
                         --v=5
 
 	# Create a Node in the API for the tests to use.
-	docker run \
+	while ! docker run \
```

Comment:
Presumably you saw this fail?  Bit odd after we successfully get the nodes above?

Reply:
Yes I think because as this node is being made and requested on the second apiserver, the second apiserver hasn't yet registered at this point so it fails with a ResourceNotFound